tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -873,6 +894,22 @@ final class TestingServer(
     }
   }
 
+  def getReferenceLocations(
+      filename: String,
+      query: String,
+      base: Map[String, String]
+  ): Future[Map[String, String]] = {
+    for {
+      (_, params) <- offsetParams(filename, query, workspace)
+      refParams = new ReferenceParams(new ReferenceContext(true))
+      _ = refParams.setPosition(params.getPosition())
+      _ = refParams.setTextDocument(params.getTextDocument())
+      references <- server.references(refParams).asScala
+    } yield {
+      TestRanges.renderLocationsAsString(base, references.asScala.toList)
```

Comment:
You can just do `assert(references.sortWith(...) == references)`.

Reply:
So I originally had that, but then I’m asserting the order is correct by the same logic I actually do the sorting with in the reference provider.

Reply:
The other option is to just have a separate assert that we can write with the expected results like:
assertReferences(
  "A.scala:5-6"
  "A.scala:8-9"
)

And translate entries to locations.

Reply:
Blah. Yea, this is frustrating that just testing the order is this difficult. I added the assertion back in right before you said this, so I will take it back out and go back to the drawing board.